### DISCUSSION TITLE
In 'group' field, `required` and `defaultValue` do nothing

### DISCUSSION THREAD

**Discussion by abaco** [payloadcms/payload#1186] [Feature Requests & Ideas]
As of version 1.1.4, fields of type `'group'` are always required, regardless of the value of the `required` property. I think this is a waste of expressiveness, especially since the semantics of an optional group field are straightforward: the whole field may be `undefined`.

For example in one of my collections I'd like to have an optional CTA with required fields. I may or may not have a CTA, but if I have one it must have both label and URL. This is how I would like to express that:
```ts
…
  {
    name: 'cta',
    type: 'group',
    fields: [
      {
         name: 'label',
         type: 'text',
         required: true,
      },
      {
         name: 'url',
         type: 'text',
         required: true,
      },
    ],
  },
…
```

But this currently requires the whole group to be defined, so it complains about undefined `label` and `url`.

The semantics of `defaultValue` are consistent with those of `required`, namely it's ignored. Now I can't come up with a good example, but I guess it might be useful to have different default values for single fields and for the whole group.

**Comment by tkrotoff** [comment]
Related discussion: https://payloadcms.com/community-help/discord/optional-group-field

> So there is no way to say "This whole thing can either be undefined, or if it exists, you are guaranteed to have these fields in it"?

There is no solution, the suggested hack _"put a condition on the group"_ doesn't work.
The data returned if cta is not filled by the user, is `cta: {}` instead of `cta: null` and doesn't match the generated TypeScript types

**Comment by DanRibbens** [comment]
As an update to this very old discussion. At some point we did add `defaultValue` support for groups.
For validation you may have to do a custom validate function, for example: 
```ts
{
    name: 'cta',
    type: 'group',
    fields: [
      {
         name: 'label',
         type: 'text',
         required: false, // this is not required so you can remove this
		 validate: (value, { siblingData }) => {
			if ( !value && siblingData.url) return 'Please provide a label for the the CTA'
            return true
		},
      },
      {
         name: 'url',
         type: 'text',
         required: false, // this is not required so you can remove this
		 validate: (value, { siblingData }) => {
			if ( !value && siblingData.label) return 'Please provide a URL for the CTA'
            return true
		},
      },
    ],
  },
```

**Reply by tkrotoff** [reply]
This is a hack for what is be a basic feature 😕

You won't have proper TypeScript types `cta?: { label: string; url: string; } | null` and the endpoint won't return what you expect
